packages/java-edition/src/json/checker/index.ts

```diff
@@ -1,18 +1,50 @@
 import type * as core from '@spyglassmc/core'
 import * as json from '@spyglassmc/json'
+import type * as mcdoc from '@spyglassmc/mcdoc'
 import { dissectUri } from '../../binder/index.js'
 
+function createTagDefinition(registry: string): mcdoc.McdocType {
+	const id: mcdoc.AttributeValue = {
+		kind: 'tree',
+		values: {
+			registry: {
+				kind: 'literal',
+				value: { kind: 'string', value: registry },
+			},
+			tags: {
+				kind: 'literal',
+				value: { kind: 'string', value: 'allowed' },
+			},
+		},
+	}
+	return {
+		kind: 'concrete',
+		child: { kind: 'reference', path: '::java::data::tag::Tag' },
+		typeArgs: [{ kind: 'string', attributes: [{ name: 'id', value: id }] }],
+	}
+}
+
 export const file: core.Checker<json.JsonFileNode> = (node, ctx) => {
 	const child = node.children[0]
 	if (ctx.doc.uri.endsWith('/pack.mcmeta')) {
-		return json.checker.definition('::java::pack::Pack')(child, ctx)
+		const type: mcdoc.McdocType = {
+			kind: 'reference',
+			path: '::java::pack::Pack',
+		}
+		return json.checker.index(type)(child, ctx)
 	}
 	const parts = dissectUri(ctx.doc.uri, ctx)
 	if (parts) {
 		if (parts?.category.startsWith('tag/')) {
-			return json.checker.resource('tag')(child, ctx)
+			const type = createTagDefinition(parts.category.slice(4))
+			return json.checker.index(type)(child, ctx)
+		}
+		const type: mcdoc.McdocType = {
+			kind: 'dispatcher',
+			registry: 'minecraft:resource',
+			parallelIndices: [{ kind: 'static', value: parts.category }],
 		}
-		return json.checker.resource(parts.category)(child, ctx)
+		return json.checker.index(type)(child, ctx)
```

Comment:
Same here for `resource` shorthand

Reply:
Dispatching `minecraft:resource` is probably only going to happen here, so I don't see why the logic should be moved to the `json` package. If anything it's the `java-edition` package that should define how the `minecraft:resource` dispatcher works.

Reply:
Well, I didn't really see the point of changing it in the first place I guess. I also don't think it's a huge issue, but I have seen a few inline mcdoc type creations in multiple places while skimming through other PRs for both `reference` and `dispatcher`. My understanding was these were neccesarry because of this change, and these were just helper functions which called `dispatcher` and `reference` in runtime, but I haven't looked too closely at it.

Reply:
I agree with Neun that those helper methods were nice